crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -545,7 +547,41 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
             }) => {
                 let transaction =
                     VerifiedTransaction::new_consensus_commit_prologue(0, 0, timestamp_ms);
-                let summary = self.execute_txn(transaction, GAS_VALUE_FOR_TESTING)?;
+                let summary = self.execute_txn(transaction, DEFAULT_GAS_BUDGET)?;
+                let output = self.object_summary_output(&summary, false);
+                Ok(output)
+            }
+            SuiSubcommand::ProgrammableTransaction(ProgrammableTransactionCommand {
+                sender,
+                gas_budget,
+                inputs,
+            }) => {
+                //  self.compiled_state().resolve_type_args(type_args)?;
+                let inputs = self.compiled_state().resolve_args(inputs)?;
+                let inputs: Vec<CallArg> = inputs
+                    .into_iter()
+                    .map(|arg| arg.into_call_arg(self))
+                    .collect::<anyhow::Result<_>>()?;
+                let file = data.ok_or_else(|| {
+                    anyhow::anyhow!("Missing commands for programmable transaction")
+                })?;
+                let contents = std::fs::read_to_string(file.path())?;
```

Comment:
Does this imply that we can only have one programmable transaction per file?  Or is the transactional runner doing something clever to split up the transactional test file into multiple chunks?

Reply:
The latter. 
The test framework breaks files at the start of commands `//#`

Reply:
<img src="https://media1.giphy.com/media/3o7TKP9ln2Dr6ze6f6/giphy.gif"/>